### ISSUE TITLE
Bootstrap 4 compatibility

### ISSUE THREAD

**Issue by DSpeichert** [OneWayTech/vue2-datatable#77]
After struggling with the `HeaderSettings` component (the settings drop down was not opening), I realized this component depends on Bootstrap 3.

With Bootstrap 4, it is not opening the drop down without ` data-toggle="dropdown"` on the button, and then with it, it's not opening the nested drop-down (for Save), it just closes the drop down completely. I'm quite sure that Bootstrap 4 does not support nested drop-downs.

**Comment by DSpeichert** [comment]
I don't think that one is related.

I fixed opening the settings drop down on this line:
https://github.com/OneWayTech/vue2-datatable/blob/b076f1b4c3b786c0b33911bb7a9ebce5c70eb22c/src/HeaderSettings/index.vue#L3
(adding `data-toggle="dropdown"`)

But then this one is broken, since it is nested:
https://github.com/OneWayTech/vue2-datatable/blob/b076f1b4c3b786c0b33911bb7a9ebce5c70eb22c/src/HeaderSettings/index.vue#L20

**Comment by kenberkeley** [comment]
well... so it does not support BS4 actually...

**Comment by DSpeichert** [comment]
No, in fact it doesn't. I'm not sure it's quite possible to support both BS3 and BS4 at the same time.
I'd recommend at least making it clear in the documentation for now that it specifically needs BS3.

**Comment by kenberkeley** [comment]
ok i will correct it tomorrow

**Comment by chadidi** [comment]
@kenberkeley Can you guys add support to BS4 ASAP, I know you can't support both BS3 and BS4 but you can make a branch for BS3 and the master for the latest bootstrap version.

Thanks

**Comment by kenberkeley** [comment]
https://github.com/OneWayTech/vue2-datatable/issues/116

**Comment by kenberkeley** [comment]
I not longer work for OneWay so you guys might have to on your own...

**Comment by chadidi** [comment]
Thanks for quick answer @kenberkeley. Who is the manager of the repository currently that can help us add BS4 support? or this repo is abandoned?

**Comment by kenberkeley** [comment]
I was the leader when in OneWay, now I left so nobody could handle... I am confident of this repo with BS3 so I will keep maintain it. In terms of BS4 compability, I hope you guys can help

**Comment by chadidi** [comment]
@kenberkeley OK I will try to do a pull request this weekend.

**Comment by vrajroham** [comment]
I managed to solve this by the following tweak, but `apply`dropdown stopped working.
```javascript
 // Added when table is rendered with little timeout
 $('div[name=HeaderSettings]').children('button[class*=dropdown-toggle]')[0].setAttribute('data-toggle','dropdown');
```

**Comment by victorhramos** [comment]
Adding to this subject, this component doesnt support bootstrap 4 tabs.
If you use inside tabs tdComp and thComp wont work.
